features/dapp_develop/dapp_develop_test.go

```diff
@@ -0,0 +1,229 @@
+      Output: colors.Colored(os.Stdout),
+      StopOnFailure: true,
+      TestingT: t,
+      DefaultContext: context.WithValue(context.Background(), e2e.TestConfigContextKey, e2eConfig) ,
+    }
+    godog.BindCommandLineFlags("godog.", opts)
+
+    err = e2e.InstallCli(e2eConfig)
+
+    if err != nil {
+       fmt.Printf("Failed to install CLI version %s, error: %v", e2eConfig.SorobanCLICrateVersion, err)
+       os.Exit(1)
+    }
+
+    status := godog.TestSuite{
+        Name:                 "soroban dapp e2e",
+        Options:              opts,
+        TestSuiteInitializer: initializeTestSuite,
+        ScenarioInitializer:  initializeScenario,
+    }.Run()
+
+    if status != 0 {
+       t.Fatal("Failed to pass all soroban dapp e2e tests")
+    }
+}
+
+func newTestConfig (e2eConfig *e2e.E2EConfig) *testConfig{
+    return &testConfig{
+        E2EConfig: e2eConfig,
+    }
+}
+
+func compileContract(ctx context.Context, contractExamplesSubPath string) error {
+    
+    testConfig := ctx.Value(e2e.TestConfigContextKey).(*testConfig)
+
+    contractWorkingDirectory := fmt.Sprintf("%s/soroban_examples", testConfig.TestWorkingDir)
+    envCmd := cmd.NewCmd("git" ,"clone", testConfig.E2EConfig.SorobanExamplesRepoURL, contractWorkingDirectory) 
+
+    status, _, err := e2e.RunCommand(envCmd, testConfig.E2EConfig)
+
+    if status != 0 || err != nil {
+        return fmt.Errorf("git clone of soroban example contracts from %s had error %v, %v", testConfig.E2EConfig.SorobanExamplesRepoURL, status, err)
+    }
+
+    envCmd = cmd.NewCmd("/bin/sh", "-c", fmt.Sprintf("cd %s; git checkout %s", contractWorkingDirectory, testConfig.E2EConfig.SorobanExamplesGitHash)) 
+
+    status, _, err = e2e.RunCommand(envCmd, testConfig.E2EConfig)
+
+    if status != 0 || err != nil {
+        return fmt.Errorf("git checkout %v of sample contracts repo %s had error %v, %v", testConfig.E2EConfig.SorobanExamplesGitHash, testConfig.E2EConfig.SorobanExamplesRepoURL, status, err)
+    }
+
+    envCmd = cmd.NewCmd("/bin/sh", "-c", fmt.Sprintf("cd %s/%s; cargo build --config net.git-fetch-with-cli=true --target wasm32-unknown-unknown --release", contractWorkingDirectory, contractExamplesSubPath)) 
+
+    status, _, err = e2e.RunCommand(envCmd, testConfig.E2EConfig)
+
+    if status != 0 || err != nil {
+        return fmt.Errorf("cargo build of sample contract %v/%v had error %v, %v", testConfig.E2EConfig.SorobanExamplesRepoURL, contractExamplesSubPath, status, err)
+    }
+
+    return nil
+}
+
+func deployContract(ctx context.Context, compiledContractFileName string) error {
+	
+    testConfig := ctx.Value(e2e.TestConfigContextKey).(*testConfig)
+    contractWorkingDirectory := fmt.Sprintf("%s/soroban_examples", testConfig.TestWorkingDir)
+
+    envCmd := cmd.NewCmd("/bin/sh", "-c", 
+        fmt.Sprintf("soroban deploy --wasm ./%s/target/wasm32-unknown-unknown/release/%s --rpc-url %s --secret-key %s --network-passphrase \"%s\"", 
```

Comment:
In go, `%q` will auto-quote strings, and escape any quotes within them. We can use that instead of `%s` here to gracefully handle paths with spaces, and network passphrases with any funny characters.
```suggestion
        fmt.Sprintf("soroban deploy --wasm %q --rpc-url %q --secret-key %q --network-passphrase %q",
          // Pre-build this so we can quote the entire filepath as one block. This means the script won't choke on any paths with spaces in them.
          fmt.Sprintf("./%s/target/wasm32-unknown-unknown/release/%s", contractWorkingDirectory, compiledContractFileName),
```

Reply:
nice, adjusted to that - https://github.com/stellar/system-test/pull/1/commits/f8f43513c1f9386d34fd50c7786c182fe044b87f